src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -85,27 +85,48 @@ bool SettingWakeUp::Refresh() {
 }
 
 void SettingWakeUp::UpdateSelected(lv_obj_t* object, lv_event_t event) {
-  if (event == LV_EVENT_VALUE_CHANGED) {
+  if (event == LV_EVENT_VALUE_CHANGED and !ignoreEvent) {
+    // Ignore all intermediate events that will be generated by checkboxes
+    // while we modify their state in this event handler.
+    ignoreEvent = true;
+    lv_checkbox_set_checked(cbOption[0], true);
     for (int i = 0; i < optionsTotal; i++) {
       if (object == cbOption[i]) {
-        lv_checkbox_set_checked(cbOption[i], true);
-
+        auto& currentCheckbox = cbOption[i];
+        auto currentCheckboxState = lv_checkbox_is_checked(currentCheckbox);
         if (i == 0) {
+          for (auto& cb : cbOption) {
+            lv_checkbox_set_checked(cb, false);
+          }
           settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::None);
-        };
-        if (i == 1) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::SingleTap);
-        };
-        if (i == 2) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap);
-        };
-        if (i == 3) {
-          settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist);
-        };
-
-      } else {
-        lv_checkbox_set_checked(cbOption[i], false);
+        } else if (i == 1) {
+          lv_checkbox_set_checked(cbOption[2], false);
+          if (currentCheckboxState) {
+            settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::SingleTap);
+          } else {
+            settingsController.unsetWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::SingleTap);
+          }
+        } else if (i == 2) {
+          lv_checkbox_set_checked(cbOption[1], false);
+          if (currentCheckboxState) {
+            settingsController.setWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap);
+          } else {
+            settingsController.unsetWakeUpMode(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap);
+          }
+        } else if (i == 3) {
```

Comment:
I know that it was like this before but replacing the magic integer literals with the relevant enum value would make this code easier to follow without needing to reference what each index refers to.

Eg. Changing
```
} else if (i == 1) {
  lv_checkbox_set_checked(cbOption[2], false);
```
to
```
} else if (i == static_cast<int>(Pinetime::Controllers::Settings::WakeUpMode::SingleTap)) {
  lv_checkbox_set_checked(cbOption[static_cast<size_t>(Pinetime::Controllers::Settings::WakeUpMode::DoubleTap)], false);
```
is easier to follow without memorizing the values of `WakeUpMode` or switching to another file to look them up, IMO at least (except maybe for all of the namespace identifiers but that's solvable too).

Reply:
Good point - but you have to add 1 to the static_casts (because the first checkbox is None).
Nevertheless, taking this advice